.github/workflows/go-lint.yml

```diff
@@ -0,0 +1,41 @@
+name: golangci-lint
```

Comment:
No need to add this workflow. It's redundant since we cover these in the build.
Agent workflows are simpler compared to Parca itself. Let's keep it so for now.